src/api/Compiler.hx

```diff
@@ -82,6 +82,25 @@ class Compiler {
 
 	}
 
+	private function isScript(source:String):Bool{
+		var resultPackage=false;
```

Comment:
it's a bit too complex, and also you `ltrim(source)` 3 times. 
Here's a simpler implementation:
```haxe
// first, ltrim the source
source = StringTools.ltrim(source);
// then check each token
for( token in ["package","import","class"] ) {
   // if source starts with either token, we know it's a script, 
   // and thus we can return true and break the loop
   if( StringTools.startsWith( source, token ) ) {
       return true;
   }
}
// otherwise, we know it's not a script
return false;
```